api/v1alpha1/httproute_types.go

```diff
@@ -181,6 +181,11 @@ type HTTPRouteAction struct {
 	// ForwardTo sends requests to the referenced object.
 	ForwardTo *core.TypedLocalObjectReference `json:"forwardTo"`
 
+	// Reencrypt specifies that the gateway controller should re-encrypt the
+	// forwarded connection.
+	// +optional
+	Reencrypt *HTTPReencrypt `json:"reencrypt"`
```

Comment:
Can we use the Service app-protocol instead of putting it here?

Reply:
@Miciah @bowei we need to make a decision on what persona is responsible for managing certificates. https://github.com/kubernetes-sigs/service-apis/pull/71/files#diff-e7d01144441bba257ce9bedd33a139c6R217 provides similar functionality but resides within `Gateway`. Is a Cluster Operator responsible for listener certs and a Dev responsible for backend certs?

@Miciah if the Dev persona is responsible for managing backend certs, then this PR is the preferred approach.